lib/providers/products_provider.dart

```diff
@@ -94,9 +96,19 @@ class ProductsProvider with ChangeNotifier {
     }
   }
 
-  ///Deletes a product in the list
-  void deleteProduct(String id) {
-    _items.removeWhere((prod) => prod.id == id);
+  ///Deletes a product in the list and in the database
+  Future<void> deleteProduct(String id) async {
+    final urlId = '${Constants.url}/products/$id.json';
+    final existingProductIndex = _items.indexWhere((prod) => prod.id == id);
+    var existingProduct = _items[existingProductIndex];
+    _items.removeAt(existingProductIndex);
     notifyListeners();
+    final response = await http.delete(urlId);
+    if (response.statusCode >= 400) {
```

Comment:
status code 400 means bad request. so does it that whenever the code is at least equal to 400, lines in this statement should run?

Reply:
yes. if the status code is equal or more than 400, the item will not be deleted and it will throw an exception that it could not delete the product.